### PR TITLE
Portal AD connection

### PR DESCRIPTION
**Pull request by spitzerr**
Connect Portal with Active directory as user store and enable automatic account creation

### CODE REVIEW

**Comment by shailesh91** [comment]
@spitzerr Each Portal needs to restart, that bit is missing. Everything else looks good.

**Comment by spitzerr** [comment]
@shailesh91 Added the restart. Even it was not necessary on our testing.

Ready to merge :-)

Fixes #42

**Comment by shailesh91** [comment]
@spitzerr I need to validate for the Portal HA (When AD is configured it should be also checked for not($Join)) and the case wherein AD connection is made post-deployment (From the looks of it Restart Only happens on the primary portal and not on the Secondary).

File: Modules/ArcGIS/ArcGISUtility.psm1 [hunk 1]
```diff
@@ -125,8 +129,12 @@ function Wait-ForServiceToReachDesiredState
     if ($Attempts++ -gt 0) {  # to skip the message for first attempt
       Write-Verbose "Checking state of Service '$ServiceName'. Attempt # $Attempts"        
     }    
+    if($MachineName -ne $null -or $MachineName -ne ""){
+        $Service = Get-Service -Name $ServiceName -ErrorAction Ignore -ComputerName $MachineName
```
Comment (shailesh91) [inline Modules/ArcGIS/ArcGISUtility.psm1]: Issue - Needs to be validated if this will work when the resource is running as a Local System Account.

**Comment by shailesh91** [comment]
@spitzerr Confirmed internally, Portal restarts are not necessary. Hence, removing it. Can you check if this looks good?

**Comment by spitzerr** [comment]
@shailesh91  Looks very good to me. On our site it`s working fine! 
Are you going to merge this one?

**Comment by shailesh91** [comment]
@spitzerr yes will merge it by the start of next week.

**Pull request merged**